packages/rlc-common/src/metadata/buildPackageFile.ts

```diff
@@ -36,391 +63,33 @@ export function buildPackageFile(model: RLCModel) {
-        packageInfo.scripts["test"] =
-          "npm run clean && tshy && npm run unit-test:node && dev-tool run bundle && npm run unit-test:browser && npm run integration-test";
-        packageInfo.scripts["unit-test:browser"] =
-          "npm run build:test && dev-tool run test:vitest --no-test-proxy --browser";
-        packageInfo.scripts["unit-test:node"] =
-          "dev-tool run test:vitest --no-test-proxy";
-        packageInfo.scripts["unit-test"] =
-          "npm run unit-test:node && npm run unit-test:browser";
-      }
-    }
-
-    if (isTypeSpecTest) {
-      // for ESM packages we use ts-node/esm loader and don't need '-r esm --require ts-node/register'
-      packageInfo["mocha"] = {
-        extension: ["ts"],
-        timeout: "1200000",
-        loader: "ts-node/esm"
-      };
-      packageInfo.scripts["unit-test:node"] =
-        'mocha --full-trace "test/{,!(browser)/**/}*.spec.ts"';
-      packageInfo.scripts["integration-test:node"] =
-        'nyc mocha --require source-map-support/register --timeout 5000000 --full-trace "dist-esm/test/{,!(browser)/**/}*.spec.js"';
-    }
-
-    packageInfo["browser"] = {
-      "./dist-esm/test/public/utils/env.js":
-        "./dist-esm/test/public/utils/env.browser.js"
-    };
-  }
+function getPackageVersion(model: RLCModel): string {
+  return model.options?.packageDetails?.version ?? "1.0.0-beta.1";
+}
 
-  if (generateSample) {
-    packageInfo["//sampleConfiguration"] = {
-      productName: model.options.serviceInfo?.title ?? model.libraryName,
-      productSlugs: ["azure"],
-      disableDocsMs: true,
-      apiRefLink: `https://docs.microsoft.com/javascript/api/${clientPackageName}${apiRefUrlQueryParameter}`
-    };
-    if (azureSdkForJs) {
-      packageInfo.scripts["execute:samples"] =
-        "dev-tool samples run samples-dev";
-    }
+function getDescription(model: RLCModel): string {
+  const description = model.options?.packageDetails?.description;
+  if (!description) {
+    return `A generated SDK for ${model.libraryName}.`;
   }
+  return description;
+}
 
-  if (flavor !== "azure") {
-    const runtimeLibVersion =
-      model.importInfo.runtimeImports.commonFallback?.version ??
-      "1.0.0-alpha.20240226.9";
-    packageInfo.dependencies = {
-      tslib: "^2.2.0",
-      "@typespec/ts-http-runtime": runtimeLibVersion
-    };
-  }
-  return packageInfo;
+function getPackageName(model: RLCModel): string {
+  return model.options?.packageDetails?.name ?? model.libraryName;
 }
 
-function appendPathWhenFormat(
-  generateTest?: boolean,
-  generateSample?: boolean
-) {
-  let path = "";
-  if (generateTest) {
-    path = path + ` "test/**/*.ts"`;
-  }
-  if (generateSample) {
-    path = path + ` "samples-dev/**/*.ts"`;
-  }
-  return path;
+function isAzureMonorepoPackage(model: RLCModel): boolean {
+  return (
+    Boolean(model.options?.azureSdkForJs) &&
+    Boolean(model.options?.flavor === "azure")
+  );
```

Comment:
The difference is on the dev depencencies. azureSdkForJs uses dev-tool and azureSdkForJs === false doesn't. In practice flavor === "azure" && azureSdkForJs  === false only would happen in our tests. It may be interesting to explore how we can remove the need to have both. Maybe taking a git submodule dependency on azure-sdk-for-js dev-tool ? Anyway, I'll explore this in a follow up effort

Reply:
makes sense to me, thanks for the detailed explaination :)